### ISSUE TITLE
Clarify licensing

### ISSUE THREAD

**Issue by voltagex** [webbju/android-plus-plus#2]
See https://news.ycombinator.com/item?id=10844817

**Comment by indrora** [comment]
The OSI explicitly says [don't just come up with one](http://opensource.org/faq#avoid-unapproved-licenses), which it looks like has happened here.

I agree that the license is... strange in how it is worded. It appears to be a modified [Zlib license](https://en.wikipedia.org/wiki/Zlib_License). 

Here's the bit that makes no sense:

```

Copyright (C) 2013-2015 NaturalMotion Limited. Used under licence. 

Code written by Justin Webb. 

Licence granted by Justin Webb with the permission of NaturalMotion Limited. 

No rights are granted by NaturalMotion Limited under this licence.

NaturalMotion Limited is not a party to this licence and has no obligations or liability under this licence.
```

I'm going to assume that Justin Webb ( @webbju ) works for NaturalMotion. Two conclusions can be drawn here:

* @webbju did this on their own time and the terms of their employment stipulate that NaturalMotion owns anything @webbju does while an employee (which has some ramifications, and [has been discussed before](https://www.reddit.com/r/gamedev/comments/1phmkr/current_employer_owns_all_software_i_develop/))
* @webbju did this on company dime and it is being released under the hand of its developer, which I find a little shady. No wait, a lot shady. GitHub has "Organizations" for this sort of thing, and at that point, NaturalMotion should own it.

(also, misspellings of license abound? UK spelling?)

**Comment by Lewiscowles1986** [comment]
Yeah, custom licenses suck, unless there is GPL with a clause for a third-party licence, but in the UK I've myself struggled to get information on licencing and software legal issues.

@indrora http://www.oxforddictionaries.com/definition/english/licence
> Note that in British English licence is the correct spelling for the noun, and is also an acceptable variant spelling of the verb. In US English both noun and verb are spelled license.

As the UK is the origin of the language, I'll defer to that :wink:

**Comment by indrora** [comment]
MPL would probably be the best option for this.

**Comment by Lewiscowles1986** [comment]
:+1: I agree it should be a standard licence, maybe one with an exception if the author is concerned about commercial projects (the current custom license doesn't read like they are, but I think they are keeping their options open)

**Comment by indrora** [comment]
It looks like the current license is Zlib, but Zlib doesn't really feel like the right kind of license for this, and it's got some extra guff that doesn't quite make sense added on top -- which makes it hard to understand.

**Comment by webbju** [comment]
Hi guys,

Sorry for the delay, I've had to carefully read the legal agreements signed with my former employer before commenting.  It's a very complicated topic as it concerns decisions made back in late-2013 when I originally signed the project permission forms.

There is scope for changing this license; but it must at all times retain the copyright notice referencing my former employer - and it must make it clear that I (as an individual) am sub-licensing with their express permission.  I don't know if anyone here has encountered anything like this before, but it might take some delicate thinking to decide on an acceptable license which meets these criteria.

Please continue to comment if you have any ideas.

**Comment by Lewiscowles1986** [comment]
:+1:  thanks for getting back to us buddy.

**Comment by indrora** [comment]
IANAL, but if the original work was built under the guise of your (former)
employer, MPL may be a good option, and it *might* be a better option for
NaturalMotion (and you) to form the "AndroidPlusPlus Contributors" --
essentially, what Android does to keep things under the right hat. You
should contact the Software Freedom Law Center to really firm up, or the
[FSFE](http://fsfe.org/index.en.html) (since you're in Europe) to just make
sure you're doing All The Right Things(tm).

Either way, *someone* needs to properly own the code. Be it you,
NaturalMotion, or someone else. Again, chat with the FSFE to make sure
things are handled correctly.
